### PR TITLE
Make setInputDevices() return a Promise that resolves with DailyDevic…

### PR DESCRIPTION
**Pull request by kompfner**
…eInfos after the input device(s) have been set.

Also, as a bonus, fix a minor bug in `getInputDevices()`.

### CODE REVIEW

File: src/module.js [hunk 1]
```diff
@@ -734,7 +734,11 @@ export default class DailyIframe extends EventEmitter {
 
     // if we're in callObject mode and not joined yet, don't do anything
     if (this._callObjectMode && this._meetingState !== DAILY_STATE_JOINED) {
-      return this;
+      return {
+        camera: { deviceId: this._preloadCache.videoDeviceId },
+        mic: { deviceId: this._preloadCache.audioDeviceId },
+        speaker: { deviceId: this._preloadCache.outputDeviceId },
+      };
     }
```
Comment (kwindla) [inline src/module.js:743]: Should we mark `setInputDevices()` as `async`, so that this gets converted to a Promise?

Comment (kompfner) [reply src/module.js:743]: whoops! good catch

**Review by kwindla — APPROVED** [review]
lgtm

**Comment by nickpell** [comment]
Very excited for this! 🎉

**Pull request merged**